src/registry.rs

```diff
@@ -102,7 +104,11 @@ impl Registry {
 
         for (index, worker) in workers.into_iter().enumerate() {
             let registry = registry.clone();
-            thread::spawn(move || unsafe { main_loop(worker, registry, index) });
+            let mut b = thread::Builder::new();
+            if let Some(name) = configuration.base_thread_name(index) {
+                b = b.name(name);
+            }
+            b.spawn(move || unsafe { main_loop(worker, registry, index) });
```

Comment:
> Now I'm surprised this isn't default-deny, but perhaps we should set that ourselves.

Yeah, we should set `#[deny(warnings)]` I guess -- it's easy to miss things when you're reviewing.